GitCommands/Settings/AppSettings.cs

```diff
@@ -1935,7 +1935,8 @@ public static void SetString(string name, string value)
             SettingsContainer.SetString(name, value);
         }
 
-        public static string GetString(string name, string? defaultValue)
+        [return: NotNullIfNotNull("defaultValue")]
+        public static string? GetString(string name, string? defaultValue)
```

Comment:
This annotation means that the return value is only null when `defaultValue` itself is null. This avoids a bunch of errors in consuming code.